### ISSUE TITLE
Exclude military building from Hero-Rookie-Settings

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#2031]
While playing some games, I came to the conclusion that at certain spots, I sometimes do not want a strategically important defense building to even have one soldier go on a stroll and leave the site halfway unprotected for more than 5 seconds. Most of the time, it takes much longer until the replacement soldier arrives at the site.

So, my suggestion would be: Make it possible (via button) to exclude a site from automatic hero-rookie exchanges. It should still be possible to send soldiers manually out of the building.

Of course, other solutions to the initial problem would also be ok, I guess... but I believe those would be harder to implement? I could imagine a way that the leaving soldier stays in the building until the arriving soldier really arrives, or such things...

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2015-09-08T06:16:39Z
 * owner: qcumber-some
 * the launchpad url was https://bugs.launchpad.net/bugs/1493215

**Comment by bunnybot** [comment]
(by kxq)
It would be straightforward to program the sites so that they request the best available soldier from start, instead of random guys like now.

It would also be easy to alter the code so that the old soldier leaves only after his replacement has arrived. This would actually simplify the code.


That said, the weakness of new military sites through soldier exchange was put into place intentionally. Your enemies have it, too, and you may take advantage. One can tackle the weakness by building a warehouse, too, and sending soldiers there in advance.

**Comment by Noordfrees** [comment]
Duplicate of #3827
